dice-game-ui-angular/src/app/services/local-game-storage-service.ts

```diff
@@ -1,33 +1,70 @@
 import { Injectable } from '@angular/core';
 import { Guid } from 'guid-typescript';
-import { GameEngine, SimpleGameEngine, Size } from 'engine';
-import { GameStorageService } from './game-storage-service';
+import { GameEngine, SimpleGameEngine } from 'engine';
+import { GameStorageService, SavedGameEngine } from './game-storage-service';
 
 @Injectable()
 export class LocalGameStorageService implements GameStorageService {
+  private readonly savedGamesKey = 'saved-games';
+
   createGame(gameEngine: GameEngine): Guid {
-    const guid = Guid.create();
-    localStorage.setItem(guid.toString(), gameEngine.fieldSize.width.toString());
-    return guid;
+    const savedGameEngine = new SavedGameEngine(gameEngine);
+    const curSavedGames = this.getAllSavedGames();
+    if (curSavedGames.filter((savedGame) => savedGame.guid === savedGameEngine.guid).length === 0) {
+      curSavedGames.push(savedGameEngine);
+      localStorage.setItem(this.savedGamesKey, JSON.stringify(curSavedGames));
+    }
+    return Guid.parse(savedGameEngine.guid);
   }
 
   saveGame(gameEngine: GameEngine, guid: Guid): Guid {
-    if (!localStorage.getItem(guid.toString())) {
-      throw new Error('The saved game was not found in the store');
+    const allSavedGames = this.getAllSavedGames();
+    if (!allSavedGames) {
+      throw new Error('This game was not created in the store.');
     }
-    localStorage.setItem(guid.toString(), gameEngine.fieldSize.width.toString());
+    const curSavedGame = allSavedGames.filter((savedGame) => savedGame.guid === guid.toString());
+    if (!curSavedGame) {
+      throw new Error('This game was not created in the store.');
+    }
+    curSavedGame[0].gameEngine = gameEngine;
+    localStorage.setItem(this.savedGamesKey, JSON.stringify(allSavedGames));
     return guid;
   }
 
   restoreGame(guid: Guid): GameEngine {
-    const fieldSize: number = JSON.parse(localStorage.getItem(guid.toString()));
-    if (!fieldSize) {
+    const allSavedGames = this.getAllSavedGames();
+    if (!allSavedGames) {
+      return null;
+    }
+    const curSavedGame = allSavedGames.filter((savedGame) => savedGame.guid === guid.toString());
+    if (!curSavedGame) {
       return null;
     }
-    return new SimpleGameEngine(fieldSize, fieldSize);
+    return new SimpleGameEngine(
+      curSavedGame[0].gameEngine.fieldSize.width,
+      curSavedGame[0].gameEngine.fieldSize.height
+    );
+  }
+
+  getAllSavedGames(): SavedGameEngine[] {
+    let curSavedGames: SavedGameEngine[] = [];
+    const savedGames = localStorage.getItem(this.savedGamesKey);
+    if (savedGames) {
+      curSavedGames = (JSON.parse(savedGames) as SavedGameEngine[]) ?? [];
+    }
+    return curSavedGames.sort((game1, game2) => (game1.creationDate < game2.creationDate ? 1 : -1));
```

Comment:
It is not this class duty to sort return games, it is a part of business logic of the application. Thus it should be done by saved games component.

Reply:
Agree. Moved to saved-game-component - the place where I need it.